MapboxNavigation/NavigationMapView.swift

```diff
@@ -885,64 +870,57 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
-                                             options: [.defaultValue : MGLConstantStyleValue<NSNumber>(rawValue: 1.5)])
+        lineCasing.lineWidth = NSExpression(format: "FUNCTION($zoomLevel, 'mgl_interpolateWithCurveType:parameters:stops:', 'linear', nil, %@)", argumentArray: [routeLineWidthAtZoomLevels.multiplied(by: 0.85)])
         
-        lineCasing.lineColor = MGLStyleValue(rawValue: routeAlternateColor)
-        lineCasing.lineCap = MGLStyleValue(rawValue: NSValue(mglLineCap: .round))
-        lineCasing.lineJoin = MGLStyleValue(rawValue: NSValue(mglLineJoin: .round))
-        lineCasing.lineOpacity = MGLStyleValue(rawValue: 0.9)
+        lineCasing.lineColor = NSExpression(forConstantValue: routeAlternateColor)
+        lineCasing.lineCap = NSExpression(forConstantValue: NSValue(mglLineCap: .round))
+        lineCasing.lineJoin = NSExpression(forConstantValue: NSValue(mglLineJoin: .round))
+        lineCasing.lineOpacity = NSExpression(forConstantValue: 0.9)
         
         return lineCasing
     }
     
     func routeWaypointCircleStyleLayer(identifier: String, source: MGLSource) -> MGLStyleLayer {
         let circles = MGLCircleStyleLayer(identifier: waypointCircleIdentifier, source: source)
-        circles.circleColor = MGLStyleValue(rawValue: UIColor(red:0.9, green:0.9, blue:0.9, alpha:1.0))
-        circles.circleOpacity = MGLStyleValue(interpolationMode: .exponential,
-                                              cameraStops: [2: MGLStyleValue(rawValue: 0.5),
-                                                            7: MGLStyleValue(rawValue: 1)],
-                                              options: nil)
-        circles.circleRadius = MGLStyleValue(rawValue: 10)
-        circles.circleStrokeColor = MGLStyleValue(rawValue: .black)
-        circles.circleStrokeWidth = MGLStyleValue(rawValue: 1)
+        circles.circleColor = NSExpression(forConstantValue: UIColor(red:0.9, green:0.9, blue:0.9, alpha:1.0))
+        circles.circleOpacity = NSExpression(format: "FUNCTION($zoomLevel, 'mgl_interpolateWithCurveType:parameters:stops:', 'linear', nil, %@)", [
+            2: 0.5,
+            7: 1])
+        circles.circleRadius = NSExpression(forConstantValue: 10)
+        circles.circleStrokeColor = NSExpression(forConstantValue: UIColor.black)
+        circles.circleStrokeWidth = NSExpression(forConstantValue: 1)
         
         return circles
     }
     
     func routeWaypointSymbolStyleLayer(identifier: String, source: MGLSource) -> MGLStyleLayer {
         let symbol = MGLSymbolStyleLayer(identifier: identifier, source: source)
         
-        symbol.text = MGLStyleValue(rawValue: "{name}")
-        symbol.textFontSize = MGLStyleValue(rawValue: 10)
-        symbol.textHaloWidth = MGLStyleValue(rawValue: 0.25)
-        symbol.textHaloColor = MGLStyleValue(rawValue: .black)
+        symbol.text = NSExpression(forConstantValue: "{name}")
```

Comment:
Pretty sure mapbox/mapbox-gl-native#10399 will prevent the legacy `{token}` syntax from working inside an expression. Use a key path expression instead: `NSExpression(forKeyPath: "name")` or `NSExpression(format: "name")`.

Reply:
Use a key path expression instead: `NSExpression(forKeyPath: "name")` or `NSExpression(format: "name")`.

/ref https://github.com/mapbox/mapbox-navigation-ios/pull/1076#discussion_r164260144